sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobOutputStreamTest.groovy

```diff
@@ -33,6 +34,27 @@ class BlobOutputStreamTest extends APISpec {
         convertInputStreamToByteArray(blockBlobClient.openInputStream()) == data
     }
 
+    @LiveOnly
+    def "BlockBlob output stream with close multiple times"() {
+        setup:
+        def data = getRandomByteArray(10 * Constants.MB)
+        def blockBlobClient = cc.getBlobClient(generateBlobName()).getBlockBlobClient()
+
+        when:
+        // set option for allowing multiple close() calls
+        def outputStream = blockBlobClient.getBlobOutputStream(new BlockBlobOutputStreamOptions(true))
+
+        outputStream.write(data)
+        outputStream.close()
+        // call again, no exceptions should be thrown
```

Comment:
I think it would be good to check the etag of the blob we're writing to and assert it doesn't change. This way we know that we're not re-committing the blob or anything. Redundant closes are just a no op in that case.